### DISCUSSION TITLE
游戏手柄功能讨论：监听手柄后有什么自动化操作需求

### DISCUSSION THREAD

**Discussion by emako** [babalae/better-genshin-impact#393] [Ideas]
手柄设置页面可参考：
https://hardwaretester.com/gamepad
https://github.com/maximglin/GamepadTester
https://github.com/alisahanyalcin/GamepadTester

API参考：
https://github.com/GenshinMatrix/Fischless/tree/main/src/Desktop/Fischless.GamepadCapture.Test

**Comment by huiyadanli** [comment]
监听可以做手柄快捷键。

手柄后续我准备搞，xinput应该可以做到后台输入

**Reply by emako** [reply]
那得用手柄模式，跟键盘不能同时用，原内部有限制

**Reply by huiyadanli** [reply]
YES，如果能够后台操作，我觉得代价是可以接受的

**Comment by huiyadanli** [comment]
发现 xinput 只能获取手柄输入状态，无法模拟输入

**Reply by emako** [reply]
https://github.com/babalae/better-genshin-impact/issues/93 输入都是靠转成键盘

**Reply by huiyadanli** [reply]
> #93 输入都是靠转成键盘

这个思路可以的，就是选择选项的时候蛋疼点。我要实现上下鼠标移动原神选项

**Comment by emako** [comment]
试了ipad接ps5手柄，游戏里选项会出现手柄选项，PC好像不行

**Reply by huiyadanli** [reply]
PC我实测要手动切换

**Comment by emako** [comment]
https://www.miyoushe.com/ys/article/37359201
